Create New changes

Olena-24 opened the pull request:
Привет, коллеги!

Я внесла несколько изменений в нашу программу управления телефонной книгой. Хочу поделиться с вами этими нововведениями:

Защита от ошибочного ввода: 

Теперь, если пользователь ошибается при вводе команды, программа предложит ему повторить ввод.  Например:

You can change:
1. NAME
2. PHONE NUMBER
3. RETURN TO MAIN MENU

Enter your text=> df

The answer doesn't exist. Please enter again

Возврат в главное меню: В каждом меню добавлена опция для возврата в главное меню. Это удобно, если пользователь случайно зашел не туда:


1. NAME
2. PHONE NUMBER
3. RETURN TO MAIN MENU

Обработка несуществующих контактов: Если пользователь пытается изменить или удалить несуществующий контакт, программа выдаст ошибку "The contact was not found" и вернет в главное меню.

Проверка правильности ввода даты рождения: При вводе даты рождения введены дополнительные проверки. Если формат даты неверный, программа уведомит об этом и предложит ввести данные заново:


Enter the year (YYYY)=> 23
Invalid year. Please enter a valid year in format YYYY. Enter the year (YYYY)=> 1983
Enter the month (MM)=> 24
Invalid month. Please enter a valid month (1-12).
Enter the month (MM)=> 03
Enter the day (DD)=> 45
Invalid day. Please enter a valid day (1-31).
Enter the day (DD)=> 12

Изменение номера телефона без ввода старого номера:  теперь в меню изменения номера телефона не требуется вводить старый номер, так как пользователь может его не помнить. Процесс выглядит так:


Enter the new phone number=> 7043769923
Phone number changed successfully to 7043769923

Надеюсь, эти изменения улучшат нашу программу и сделают ее более удобной и надежной для использования.

Olena-24 committed:
Create New changes

Привет, коллеги!

Я внесла несколько изменений в нашу программу управления телефонной книгой. Хочу поделиться с вами этими нововведениями:

Защита от ошибочного ввода: 

Теперь, если пользователь ошибается при вводе команды, программа предложит ему повторить ввод. 
Например:

You can change:
1. NAME
2. PHONE NUMBER
3. RETURN TO MAIN MENU

Enter your text=> df

The answer doesn't exist. Please enter again

Возврат в главное меню: В каждом меню добавлена опция для возврата в главное меню. Это удобно, если пользователь случайно зашел не туда:


1. NAME
2. PHONE NUMBER
3. RETURN TO MAIN MENU

Обработка несуществующих контактов: Если пользователь пытается изменить или удалить несуществующий контакт, программа выдаст ошибку "The contact was not found" и вернет в главное меню.

Проверка правильности ввода даты рождения: При вводе даты рождения введены дополнительные проверки. Если формат даты неверный, программа уведомит об этом и предложит ввести данные заново:


Enter the year (YYYY)=> 23
Invalid year. Please enter a valid year in format YYYY.
Enter the year (YYYY)=> 1983
Enter the month (MM)=> 24
Invalid month. Please enter a valid month (1-12).
Enter the month (MM)=> 03
Enter the day (DD)=> 45
Invalid day. Please enter a valid day (1-31).
Enter the day (DD)=> 12

Изменение номера телефона без ввода старого номера: 
теперь в меню изменения номера телефона не требуется вводить старый номер, так как пользователь может его не помнить. Процесс выглядит так:


Enter the new phone number=> 7043769923
Phone number changed successfully to 7043769923

Надеюсь, эти изменения улучшат нашу программу и сделают ее более удобной и надежной для использования.

Olena-24 committed:
Update New changes

Привет, коллеги!

Я внесла несколько изменений в нашу программу управления телефонной книгой. Хочу поделиться с вами этими нововведениями:

Защита от ошибочного ввода: Теперь, если пользователь ошибается при вводе команды, программа предложит ему повторить ввод. Например:


You can change:
1. NAME
2. PHONE NUMBER
3. RETURN TO MAIN MENU
Enter your text=> df
The answer doesn't exist. Please enter again

Возврат в главное меню: В каждом меню добавлена опция для возврата в главное меню. Это удобно, если пользователь случайно зашел не туда:

1. NAME
2. PHONE NUMBER
3. RETURN TO MAIN MENU
Обработка несуществующих контактов: Если пользователь пытается изменить или удалить несуществующий контакт, программа выдаст ошибку "The contact was not found" и вернет в главное меню.

Проверка правильности ввода даты рождения: При вводе даты рождения введены дополнительные проверки. Если формат даты неверный, программа уведомит об этом и предложит ввести данные заново:

Enter your text=> 1
Enter the year (YYYY)=> 23
Invalid year. Please enter a valid year in format YYYY.
Enter the year (YYYY)=> 1983
Enter the month (MM)=> 24
Invalid month. Please enter a valid month (1-12).
Enter the month (MM)=> 03
Enter the day (DD)=> 45
Invalid day. Please enter a valid day (1-31).
Enter the day (DD)=> 12

Изменение номера телефона без ввода старого номера: Теперь в меню изменения номера телефона не требуется вводить старый номер, так как пользователь может его не помнить. Процесс выглядит так:

Enter the new phone number=> 7043769923
Phone number changed successfully to 7043769923
Надеюсь, эти изменения улучшат нашу программу и сделают ее более удобной и надежной для использования.

Olena-24 commented:
так же обновила работу delete. пользователь может делать поиск контакта для удаления  по дню рождения и номеру телефона и программа у него забрасывает подтверждение хочет он удалить или нет

SkyFoxNa commented:
Не коректно выводит полную информацию по заданному кол-во контактов
Выводит <generator object AddressBook.iterator at 0x7f26de1a57a0>
Могу предложить мой код, он универсальный как на одно кол-во так и на все записи.
Если надо определенно кол-во то передаем item_number если же нет, то он возмет по умолчанию 1000

    def iterator(self, item_number=1000):
        for i in range(0, len(self.data), item_number):
            yield list(self.data.values())[i:i+item_number]

pentagon846 commented:
так же очень много потверждений нужно вводить по типу :Do you want to display all contacts? Please enter the number:
лучше сделать только на критические пункты.

Svetilnikov committed:
виправлено вивід Birthday

Привіт, виправив помилку виводу інфи по контакту, додав вивід поля birthday, до того виводилась тільки кількість днів, також виправив змінив 'Day' на 'Days'

тобто

було:
|| Name: Andrii  || Phones: 0634996203; || Day to birthday: 233

стало:
|| Name: Andrii  || Phones: 0634996203; || Birthday: 1982.08.06 || Days to birthday: 233||
